spec/requests/mutations/resend_confirmation_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe 'Resend confirmation' do
+  include_context 'with graphql query request'
+
+  let(:user)     { create(:user, confirmed_at: nil) }
+  let(:email)    { user.email }
+  let(:id)       { user.id }
+  let(:redirect) { Faker::Internet.url }
+  let(:query) do
+    <<-GRAPHQL
+      mutation {
+        userResendConfirmation(
+          email:"#{email}",
+          redirectUrl:"#{redirect}"
+        ) {
+          message
+          authenticable {
+            id
+            email
+          }
+        }
+      }
+    GRAPHQL
+  end
+
+  context 'when params are correct' do
+    it 'sends an email to the user with confirmation url and returns a success message' do
+      expect { post_request }.to(change(ActionMailer::Base.deliveries, :count).by(1))
+      expect(json_response[:data][:userResendConfirmation]).to include(
+        authenticable: {
+          id: id,
+          email: email
+        },
+        message: "You will receive an email with instructions for how to confirm your email address in a few minutes."
+      )
+      
+      email = Nokogiri::HTML(ActionMailer::Base.deliveries.last.body.encoded)
+      link  = email.css('a').first
+      confirm_link_msg_text = email.css('p')[1].inner_html
+      confirm_account_link_text = link.inner_html
+
+      expect(confirm_link_msg_text).to eq("You can confirm your account email through the link below:")
+      expect(confirm_account_link_text).to eq("Confirm my account")
+
+      # TODO: Move to feature spec
+      expect do
+        get link['href']
+        user.reload
+      end.to change(user, :confirmed_at).from(NilClass).to(ActiveSupport::TimeWithZone)
+    end
+
+    context 'when the user has already been confirmed' do
+      before { user.confirm }
+
+      it 'does *NOT* send an email and raises an error' do
+        expect { post_request }.to(change(ActionMailer::Base.deliveries, :count).by(0))
```

Comment:
Here is the line @00dav00 was talking about.

```ruby
expect { post_request }.to not_change(ActionMailer::Base.deliveries, :count)
```
I think it is also preferred not to used parenthesis on the `to` method as a convention

Reply:
I guess the parenthesis thing is an rspec convention, please refer to other specs to make spec style consistent

Reply:
No, this makes sense. Its a kind of "rubism" also. I think I read it somewhere - maybe why's guide to ruby or something. I've fixed this and went through my rspec file and looked for violations of this standard also.